webservices/resources/candidates.py

```diff
@@ -1,290 +1,131 @@
-    'active_through': fields.Integer,
-    'election_years': fields.List(fields.Integer),
-    'incumbent_challenge_full': fields.String,
-    'incumbent_challenge': fields.String,
-    'office_full': fields.String,
-    'office': fields.String,
-    'party_full': fields.String,
-    'party': fields.String,
-    'state': fields.String,
-    'name': fields.String,
-}
-candidate_detail_fields = {
-    'candidate_id': fields.String,
-    'candidate_status_full': fields.String,
-    'candidate_status': fields.String,
-    'district': fields.String,
-    'active_through': fields.Integer,
-    'election_years': fields.List(fields.Integer),
-    'incumbent_challenge_full': fields.String,
-    'incumbent_challenge': fields.String,
-    'office_full': fields.String,
-    'office': fields.String,
-    'party_full': fields.String,
-    'party': fields.String,
-    'state': fields.String,
-    'name': fields.String,
-    'expire_date': fields.String,
-    'load_date': fields.String,
-    'form_type': fields.String,
-    'address_city': fields.String,
-    'address_state': fields.String,
-    'address_street_1': fields.String,
-    'address_street_2': fields.String,
-    'address_zip': fields.String,
-    'candidate_inactive': fields.String,
-    'committees': fields.Nested(candidate_commitee_fields),
-}
-candidate_history_fields = {
-    'candidate_id': fields.String,
-    'two_year_period': fields.Integer,
-    'candidate_status_full': fields.String,
-    'candidate_status': fields.String,
-    'district': fields.String,
-    'incumbent_challenge_full': fields.String,
-    'incumbent_challenge': fields.String,
-    'office_full': fields.String,
-    'office': fields.String,
-    'party_full': fields.String,
-    'party': fields.String,
-    'state': fields.String,
-    'name': fields.String,
-    'expire_date': fields.String,
-    'load_date': fields.String,
-    'form_type': fields.String,
-    'address_city': fields.String,
-    'address_state': fields.String,
-    'address_street_1': fields.String,
-    'address_street_2': fields.String,
-    'address_zip': fields.String,
-    'candidate_inactive': fields.String,
-}
-pagination_fields = {
-    'per_page': fields.Integer,
-    'page': fields.Integer,
-    'count': fields.Integer,
-    'pages': fields.Integer,
-}
-candidate_list_fields = {
-    'api_version': fields.Fixed(1),
-    'pagination': fields.Nested(pagination_fields),
-    'results': fields.Nested(candidate_fields),
+from sqlalchemy.sql import text, or_
+from flask.ext.restful import Resource
+
+from webservices import args
+from webservices import spec
+from webservices import paging
+from webservices import schemas
+from webservices.common.util import filter_query
+from webservices.common.models import db, Candidate, CandidateDetail, CandidateHistory, CandidateCommitteeLink
+
+
+filter_fields = {
+    'candidate_id',
+    'candidate_status',
+    'district',
+    'incumbent_challenge',
+    'office',
+    'party',
+    'state',
```

Comment:
We need a year filter for `/candidates` is this list just for detail and history?